stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1291,6 +1290,14 @@ void DiscardingTaskGroup::offer(AsyncTask *completedTask, AsyncContext *context)
       // This is the last task, we have a waiting task and there was no error stored previously;
       // We must resume the waiting task with a success, so let us return here.
       resumeWaitingTask(completedTask, assumed, /*hadErrorResult=*/false, alreadyDecrementedStatus);
+
+//      // TODO: since the DiscardingTaskGroup ended up written as `-> T` in order to use the same pointer auth as the
+//      //       usual task closures; we end up retaining the value when it is returned. As this is a discarding group
+//      //       we actually can and should release this value.
+//      //       Is there a way we could avoid the retain made on the returned value entirely?
+//      if (completedTask->futureFragment()->getResultType()->isClassObject()) {
+//        swift_release(reinterpret_cast<HeapObject *>(completedTask->futureFragment()->getStoragePtr()));
+//      }
```

Comment:
This is not what we want; instead we want to fix the signature of the group addTask